Feature/http client via messagingservice

gunther-bachmann opened the pull request.

gunther-bachmann committed:
reworked test execution service to use messagingservice to get Httpclient

gunther-bachmann committed:
fixed all tests

gunther-bachmann committed:
fix compile errors

Commit:
Revert "move webdriver manager update into earlier itest stage"

This reverts commit 2ee195963afbdb93ede3620c950e2ff28870953d.

gunther-bachmann committed:
cleanup, documentation, added dev dependencies

jjelschen commented on integration/e2e/app.e2e-spec.ts:
```diff
@@ -23,7 +23,10 @@ describe('Workspace Navigator tests', function () {
        if (window.module != "aot") { \
-         result = parseInt(window.getComputedStyle(document.getElementById("new-file", null)).getPropertyValue("height"), 10); \
+         var element = document.getElementById("new-file", null); \
+         if (element) { \
```
shouldn't the test fail otherwise? Or is the absence of 'new-file' somehow expected?

gunther-bachmann replied:
true, had to fix some aot setup to make this test work again

Commit:
disable gecko driver download

workaround to avoid hitting GitHub's rate limit, see https://github.com/angular/webdriver-manager/issues/303

jjelschen commented on package.json:
```diff
@@ -96,9 +96,8 @@
-  "dependencies": {
-    "yarn": "^1.5.1"
+    "zone.js": "0.8.20",
+    "webdriver-manager": "^12.0.6",
```
That's probably only necessary because I messed up the integration test setup sometime earlier, to work around an issue we had with the integration tests on Travis, which, when run in parallel for different browsers, ran into GitHub's rate limit (because the Gecko driver is downloaded from GitHub…). I partly reverted those changes on [this branch](https://github.com/test-editor/web-workspace-navigator/tree/fix/revert-webdriver-move); not sure whether we should use that, or just introduce the webdriver-manager dependency in the parent directory, as you have.

gunther-bachmann replied:
merged your branch into this ...

jjelschen commented on src/lib/src/component/tree-viewer/new-element.component.spec.ts:
```diff
@@ -1,7 +1,7 @@
 import { By } from '@angular/platform-browser';
-import { async, TestBed, ComponentFixture } from '@angular/core/testing';
-import { mock, instance, verify, when, anyString } from 'ts-mockito';
+import { async, TestBed, ComponentFixture, tick } from '@angular/core/testing';
```
I don't think 'tick' is actually used anywhere (since you do not also import its prerequisite, 'fakeAsync')

jjelschen commented on src/lib/src/component/tree-viewer/tree-viewer.component.spec.ts:
```diff
@@ -1,8 +1,8 @@
-import { async, ComponentFixture, TestBed, inject } from '@angular/core/testing';
+import { async, ComponentFixture, TestBed, inject, fakeAsync } from '@angular/core/testing';
```
… and here you're not using fakeAsync :)

jjelschen commented on src/lib/src/service/execution/test.execution.service.spec.ts:
```diff
@@ -1,30 +1,44 @@
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
```
uses fakeAsync, but never uses tick, so the former is probably unnecessary, as well.

jjelschen commented on src/lib/src/service/execution/test.execution.service.ts:
```diff
@@ -17,9 +23,9 @@ export interface TestExecutionStatus {
-  abstract execute(path: string): Promise<any>;
-  abstract getStatus(path: string): Promise<TestExecutionStatus>;
-  abstract getAllStatus(): Promise<TestExecutionStatus[]>;
+  abstract execute(path: string, onThen?: (some: any) => void, onError?: (error: any) => void): void
```
just if its low-impact: consider renaming "onThen" to "onResponse" (just think the former sounds a bit weird…)

jjelschen commented on src/lib/src/service/execution/test.execution.service.ts:
```diff
@@ -31,38 +37,40 @@ export class DefaultTestExecutionService extends TestExecutionService {
-    return this.getHttpClient().post(this.getURL(path, DefaultTestExecutionService.executeURLPath), '').toPromise();
+  getStatus(path: string,
+            onThen?: (status: TestExecutionStatus) => void,
+            onError?: (error: any) => void): void { // Promise<TestExecutionStatus> {
```
remove out-commented return type?

jjelschen commented on src/lib/src/service/execution/test.execution.service.ts:
```diff
@@ -31,38 +37,40 @@ export class DefaultTestExecutionService extends TestExecutionService {
-  getAllStatus(): Promise<TestExecutionStatus[]> {
-    return this.getHttpClient().get<any[]>(`${this.serviceUrl}${DefaultTestExecutionService.statusAllURLPath}`).toPromise().then(statusResponse => {
+  getAllStatus(onThen?: (status: TestExecutionStatus[]) => void,
+               onError?: (error: any) => void): void { //  Promise<TestExecutionStatus[]> {
```
remove out-commented return type?

jjelschen commented on src/lib/src/component/tree-viewer/new-element.component.spec.ts:
```diff
@@ -162,18 +162,21 @@ describe('NewElementComponent', () => {
 
+    // and given that
+    const [path, typeString, onSuccess, onError] = capture(persistenceService.createResource).last();
+    onSuccess.apply(null, ['some/path']);
```
is this different from just calling `onSuccess(null, ['some/path'])`?

gunther-bachmann committed:
fix integration:aot

gunther-bachmann committed:
review based fixes

gunther-bachmann committed:
Merge remote-tracking branch 'origin/fix/revert-webdriver-move' into feature/http_client_via_messagingservice

gunther-bachmann committed:
more review based changes

gunther-bachmann committed:
Merge remote-tracking branch 'origin/master' into feature/http_client_via_messagingservice

gunther-bachmann committed:
replace onThem with onResponse

gunther-bachmann committed:
remove Observables, since they are not needed
